nixos/modules/services/misc/podgrab.nix

```diff
@@ -0,0 +1,69 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.podgrab;
+in
+{
+  options.services.podgrab = with lib; {
+    enable = mkEnableOption "Podgrab, a self-hosted podcast manager";
+
+    user = mkOption {
+      type = types.str;
+      default = "podgrab";
+      description = "User account under which podgrab runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "podgrab";
+      description = "Group under which podgrab runs.";
+    };
+
+    passwordFile = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "/run/secrets/password.env";
+      description = ''
+        The path to a file containing the PASSWORD environment variable
+        definition for Podgrab's authentification.
+      '';
+    };
+
+    openFirewall = mkEnableOption "opening of the webui port in the firewall";
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.podgrab = {
+      wantedBy = [ "multi-user.target" ];
+      preStart = ''
+        mkdir -p /var/lib/podgrab/
+        rm -f /var/lib/podgrab/{client,webassets}
+        ln -s ${pkgs.podgrab}/share/{client,webassets} /var/lib/podgrab/
+      '';
+      serviceConfig = {
+        EnvironmentFile = lib.optional (cfg.passwordFile != null) [
+          cfg.passwordFile
+        ];
+        ExecStart = "${pkgs.podgrab}/bin/podgrab";
+        WorkingDirectory = "/var/lib/podgrab/";
+        StateDirectory = "podgrab";
+        User = cfg.user;
+        Group = cfg.group;
+      };
+    };
+
+    users.users = lib.mkIf (cfg.user == "podgrab") {
```

Comment:
Can you use a systemd DynamicUser for this (see https://github.com/NixOS/rfcs/blob/master/rfcs/0052-dynamic-ids.md) instead of creating a user and group?

Reply:
Sure. I figured that it could be useful to change the group/user for permission: for example a common `nas` group used by services that manage media files.

Reply:
Sure! DynamicUser works well with groups; you can set a Group attribute on the systemd unit and systemd will make sure the user (named, if necessary) is a member of the group for the lifetime of the process.

I don't actually know what systemd does if your named User already exists; if someone explicitly specifies a user it probably makes sense to turn DynamicUser off.